pkg/artifactory/resource_xray_policy.go

```diff
@@ -442,6 +442,11 @@ func resourceXrayPolicyRead(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).Xray
 
 	policy, resp, err := c.V1.Policies.GetPolicy(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.